libs/x-content/src/test/java/org/elasticsearch/common/xcontent/XContentParserTests.java

```diff
@@ -329,6 +330,54 @@ public void testNestedMapInList() throws IOException {
         }
     }
 
+    public void testGenericMap() throws IOException {
+        String content = "{" +
+            "\"a\": { \"i\": 1, \"d\": 0.1, \"s\": \"aaa\" }, " +
+            "\"b\": { \"i\": 2, \"d\": 0.2, \"s\": \"bbb\" }, " +
+            "\"c\": { \"i\": 3, \"d\": 0.3, \"s\": \"ccc\" }" +
+            "}";
+        Map<String, SimpleStruct> expectedMap =
+            Map.of(
+                "a", new SimpleStruct(1, 0.1, "aaa"),
+                "b", new SimpleStruct(2, 0.2, "bbb"),
+                "c", new SimpleStruct(3, 0.3, "ccc"));
+        try (XContentParser parser = createParser(JsonXContent.jsonXContent, content)) {
+            assertThat(parser.genericMap(SimpleStruct::fromXContent), equalTo(expectedMap));
+            assertNull(parser.nextToken());
+        }
+    }
+
+    public void testGenericMapOrdered() throws IOException {
+        String content = "{" +
+            "\"a\": { \"i\": 1, \"d\": 0.1, \"s\": \"aaa\" }, " +
```

Comment:
This obviously works fine, but I wonder if we could make the test more clear if we use a reverse order here
The order a,b,c would be maintained even if it method `genericMap` (not order) was used.
WDYT?